protos/rowids.proto

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+syntax = "proto3";
+
+package lance.table;
+// TODO: what would it take to store this in a LanceV2 file?
+// Or would flatbuffers be better for this?
```

Comment:
Leaving this TODO for a future PR. Would appreciate input on how we want to support this. For now, protobufs seems like the easiest.

Reply:
> less with lance v2 and more with Arrow

Yeah, that's what I was thinking too. I think it's likely very important we keep the in-memory and on-disk format aligned to minimize serialization.